Shopify/vscode-as
# Automatic sensible defaults

This PR introduces the following features:

- Automatic sensible defaults for AssemblyScript sources, as of this PR there's no longer a need to disable typescript validation in the entire project to make `asls` work correctly. This will be handled automatically. More importantly, typescript validations will still work if the current file doesn't belong to the specified `include` configuration key.
- Swap the `root` configuration key with `include` this makes it clearer that we are dealing with a glob pattern rather than a directory.


I didn't add unit tests to the server plugin or the to wrapper plugin module given that it will require mocking quite a lot of pieces, and the ROI doesn't seem worth it. This is better suited for an integration test IMO. For now, testing manually should do it.

To test:

- Clone this PR
- Run the extension
- Ensure that you **don't** have in `.vscode/settings` of your test project
```
# in .vscode/settings.json
{
  "typescript.validate.enable": false
}
```
- Go to your any source in your `assembly/` directory and verify that the surfaced diagnostics only come from `asls`.
- Go to an arbitrary source outside the `assembly` directly make sure that the surfaced diagnostics come from the ts language server. There's an issue with TS sources in which `asls` will still analyze them and emit diagnostics for them, causing multiple diagnostics to be surfaced (from the ts server and from `asls`) I'll fix this as a follow-up PR in the server to ensure that only files that comply with the glob are analyzed.
